### PR TITLE
Added ``FUNC_NETWORK_UP`` and ``FUNC_NETWORK_DOWN`` events

### PR DESCRIPTION
**Pull request by s-hadinger**
## Description:

Added `FUNC_NETWORK_UP` and `FUNC_NETWORK_DOWN` events occuring when either Wifi or ETH network comes up. This is the moment when drivers can listen to TCP/UDP ports.

Started to move KNX and HUE/WEMO out of `support.ino` to respond to those new events.

## Checklist:
  - [x] The pull request is done against the latest development branch
  - [x] Only relevant files were touched
  - [x] Only one feature/fix was added per PR and the code change compiles without warnings
  - [x] The code change is tested and works with Tasmota core ESP8266 V.2.7.4.9
  - [x] The code change is tested and works with Tasmota core ESP32 V.2.0.5
  - [x] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

### CODE REVIEW

File: tasmota/tasmota_support/support_tasmota.ino [hunk 1]
```diff
@@ -1569,76 +1569,81 @@ void Every250mSeconds(void)
...
-      if (!knx_started && Settings->flag.knx_enabled) {  // CMND_KNX_ENABLED
-        KNXStart();
-        knx_started = true;
-      }
-#endif  // USE_KNX
+        // send FUNC_NETWORK_UP to all modules
+        if (network_state_changed) {
+          // AddLog(LOG_LEVEL_DEBUG_MORE, PSTR("WIF: Sending FUNC_NETWORK_UP"));
+          XdrvCall(FUNC_NETWORK_UP);
+          XsnsCall(FUNC_NETWORK_UP);
```
Comment (arendst) [inline tasmota/tasmota_support/support_tasmota.ino:1630]: Pls use new function XdrvXsnsCall(FUNC_NETWORK_UP) replacing the two.

File: tasmota/tasmota_support/support_tasmota.ino [hunk 2]
```diff
@@ -1569,76 +1569,81 @@ void Every250mSeconds(void)
...
 #endif  // USE_DEVICE_GROUPS
 
-#ifdef USE_KNX
-      knx_started = false;
-#endif  // USE_KNX
+        // send FUNC_NETWORK_UP to all modules
+        if (network_state_changed) {
+          // AddLog(LOG_LEVEL_DEBUG_MORE, PSTR("WIF: Sending FUNC_NETWORK_DOWN"));
+          XdrvCall(FUNC_NETWORK_DOWN);
+          XsnsCall(FUNC_NETWORK_DOWN);
```
Comment (arendst) [inline tasmota/tasmota_support/support_tasmota.ino:1644]: Pls use new function XdrvXsnsCall(FUNC_NETWORK_UP) replacing the two.

File: tasmota/tasmota_support/support_tasmota.ino (hunk 1 above)
Comment (s-hadinger) [reply tasmota/tasmota_support/support_tasmota.ino]: Done

File: tasmota/tasmota_support/support_tasmota.ino (hunk 2 above)
Comment (s-hadinger) [reply tasmota/tasmota_support/support_tasmota.ino]: Done

**Pull request merged**